apps/api/src/vault/vault.controller.ts

```diff
@@ -0,0 +1,74 @@
+import { Controller, Get, Post, Body, Patch, Param, Delete, UnauthorizedException } from '@nestjs/common'
+import { AuthenticatedUser } from 'nest-keycloak-connect'
+
+import { VaultService } from './vault.service'
+import { CreateVaultDto } from './dto/create-vault.dto'
+import { UpdateVaultDto } from './dto/update-vault.dto'
+import { KeycloakTokenParsed } from '../auth/keycloak'
+
+@Controller('vault')
+export class VaultController {
+  constructor(private readonly vaultService: VaultService) { }
+
+  @Post()
+  create(@AuthenticatedUser() user: KeycloakTokenParsed,
+    @Body() createVaultDto: CreateVaultDto) {
+    if (!user) {
+      throw new UnauthorizedException()
+    }
+
+    return this.vaultService.create(createVaultDto)
+  }
```

Comment:
For this and all methods changing the vault: Only the coordinator can create or change Vaults. So, here we have to check the user.id with the coordinatorID of the campaign.